### PR TITLE
Feat: switch off c02 parameters

### PR DESCRIPTION
**Pull request by dayantur**
Adressing issue #889 - when `emissionsmethod` is set to 0–4, nullify CO2-related site parameters (including day/hourly profiles and "alpha_bioco2", "alpha_enh_bioco2", "beta_bioco2", "beta_enh_bioco2", "min_res_bioco2", "theta_bioco2") so CO2 parameters do not trigger Pydantic checks.

### Main Changes
- human_activity.py and profile.py: DayProfile and HourlyProfile fields to be optional
- "alpha_bioco2", "alpha_enh_bioco2", "beta_bioco2", "beta_enh_bioco2", "min_res_bioco2", "theta_bioco2" in site.py to be optional
- profile.py: field_validator to accept none
- yaml_helpers.py: nullify co2 and *_bioco2 params if emissionsmethod 0..4
- phase_b.py: save nullified params into report

### Tests
- Added suite of tests

### CODE REVIEW

File: src/supy/data_model/validation/pipeline/phase_b.py [hunk 1]
```diff
@@ -1381,6 +1382,87 @@ def _recursive_nullify(block: dict, path: str = ""):
...
+            co2_block = anth_emis.get("co2", {})
+            site_gridid = get_site_gridid(site)
+
+            if co2_block:
+                nullified_params = []
+
+                def _recursive_nullify_co2(block: dict, path: str = ""):
+                    """
+                    Recursively nullify CO2-related parameters.
+
```
Comment (sunt05) [inline src/supy/data_model/validation/pipeline/phase_b.py]: **Inconsistent None-check**

In yaml_helpers.py line 1263, there's an explicit guard:
```python
if emissionsmethod is not None and emissionsmethod in (0, 1, 2, 3, 4):
```

But here there's no `is not None` check. While `None in [0,1,2,3,4]` is False, adding the explicit check would be more defensive and consistent:
```python
if emissionsmethod is not None and emissionsmethod in [0, 1, 2, 3, 4]:
```

File: src/supy/data_model/core/human_activity.py [hunk 2]
```diff
@@ -373,7 +373,7 @@ class CO2Params(BaseModel):  # TODO: May need to add the RefValue to the profile
             "display_name": "Energy Emission Factor Vehicles",
         },
     )
-    fcef_v_kgkm: DayProfile = Field(
+    fcef_v_kgkm: Optional[DayProfile] = Field(
```
Comment (sunt05) [inline src/supy/data_model/core/human_activity.py]: **Semantic note**

Field is `Optional[DayProfile]` but has `default_factory=DayProfile`, so it will never actually be `None` unless explicitly set during validation preprocessing.

This is valid but may confuse future developers. Consider adding a brief comment explaining that `Optional` is needed because the field may be nullified when emissionsmethod is 0-4.

File: src/supy/data_model/core/profile.py [hunk 3]
```diff
@@ -157,8 +159,8 @@ class HourlyProfile(BaseModel):
 
     model_config = ConfigDict(title="Hourly Profile")
 
-    working_day: Dict[str, float]
-    holiday: Dict[str, float]
+    working_day: Optional[Dict[str, Optional[float]]] = None
```
Comment (sunt05) [inline src/supy/data_model/core/profile.py]: **Potential breaking change**

`HourlyProfile.working_day` and `holiday` previously had explicit dict defaults from `__init__`; now they default to `None`.

Existing code that expects these to always be dicts may break. Please verify no existing code assumes these are always dicts, or consider using `default_factory` to maintain backwards compatibility if needed.

File: test/data_model/test_yaml_processing.py [hunk 4]
```diff
@@ -1932,6 +1932,26 @@ def test_stebbsmethod1_leaves_stebbs_untouched():
...
+    }
+    data = {
+        "model": {"physics": {"emissionsmethod": 0}},  # < 5 => CO2 disabled
+        "sites": [_build_site_with_co2(co2_block)],
+    }
+    result = precheck_model_option_rules(deepcopy(data))
+
+    out = result["sites"][0]["properties"]["anthropogenic_emissions"]["co2"]
+    assert out["co2pointsource"]["value"] is None
+    assert out["nested"]["ef_umolco2perj"]["value"] is None
```
Comment (sunt05) [inline test/data_model/test_yaml_processing.py]: **Consider additional test coverage**

This test covers the basic happy path. Consider adding tests for:
- Empty `co2_block`
- `emissionsmethod` is `None`
- Deeply nested structures in CO2 block
- The phase_b.py `adjust_model_dependent_nullification()` code path
- Verifying the profile model changes accept `None` values correctly

**Review by sunt05 — COMMENTED** [review]
## Code Review Summary

Overall good implementation following established patterns. A few items to address before merging.

**Key concerns:**
1. Duplicate nullification logic between yaml_helpers.py and phase_b.py
2. Inconsistent None-check guards
3. Limited test coverage for edge cases

See inline comments for details.

**Comment by dayantur** [comment]
Hi @sunt05  - thanks for the review :)

> 1. **Consolidate or document the two recursive nullify functions** — clarify why both `recursive_nullify_any()` and `_recursive_nullify_co2()` are needed
> 2. **Add None-check guard in phase_b.py:1403** — to match yaml_helpers.py's defensive check


1. These are slightly different, but agree that we can make them more concise/eliminate part of the duplicates. I tried a strategy before changing the PR to ready for review, but was unsucessful. I have another idea now, so I will try to implement that (basically, the nullifying function in phase_b is doing the same job as the one in the yaml_helper, but also adding the nullified params to the report - so in principle the two should be merged into a single one to store in yaml_helper)

2. Yes - that makes sense

> 
> **Quick questions:**
> 
> * Is the `WeeklyProfile` Optional change intentional? It doesn't seem directly related to CO2 nullification.

It is intentional - but I might test again if really needed. When working with the STEBBS nullification logic, I checked the code to be sure that the STEBBS parameters could be null, i.e. Optional (otherwise Pydantic was raising them as error). I haven't tried again in this case, but will do to see if we really need the Optional also here

> * Should the phase_b.py nullification mirror precheck exactly, or are the behavioural differences intentional?
> 
> Happy to discuss any of these!

Answered above :)

Will make these changes probably early next week ;)
